src/main/java/com/cryptomorin/xseries/skull/XSkull.java

```diff
@@ -439,6 +631,13 @@ private static GameProfile fetchProfile(GameProfile profile) {
         return sanitizeProfile(profile);
     }
 
+    /**
+     * Sanitizes the provided {@link GameProfile} by removing unnecessary timestamp data
+     * and caches the sanitized profile.
+     *
+     * @param profile The {@link GameProfile} to be sanitized.
+     * @return The sanitized {@link GameProfile}.
+     */
     @SuppressWarnings("deprecation")
     private static GameProfile sanitizeProfile(GameProfile profile) {
         return Optional.ofNullable(Iterables.getFirst(profile.getProperties().get("textures"), null))
```

Comment:
I don't think that the functional approach really suits this situation here. I prefer the traditional imperative style here.